lib/cache/storages/factory.ts

```diff
@@ -0,0 +1,22 @@
+import { Factory } from '../../interfaces/factory';
+import { MemoryStorage } from './MemoryStorage';
+import { Storage } from './Storage';
+
+export class StorageFactory implements Factory<Storage> {
+
+  constructor(
+    private readonly limit: number,
+    private readonly storage: 'memory',
+  ) { }
+
+  public create(): Storage {
```

Comment:
I'd recommend that the `storage` type is passed as an argument to the `create` function instead of in constructor.